src/sessionManager.ts

```diff
@@ -45,9 +45,33 @@ export default function SessionManager(
                 self.endSession();
                 self.startNewSession();
             } else {
+                const user = mpInstance.Identity.getCurrentUser()
+                let needToIdentify = false
+
+                if (user) {
+                    // If userIdentities is an empty object but identifyRequest
```

Comment:
I think from an organizational standpoint, you should move this into our helpers.js or identity.js module, or make a new method within the SessionManager called `needsToIdentify`.

It would make testing easier and make things more compartmentalized.

Reply:
I would prefer to keep this in TypeScript instead of JS so I'll go with your second suggestion 👍